### DISCUSSION TITLE
Function to transform confidence scores to logit

### DISCUSSION THREAD

**Discussion by GrunCrow** [ultralytics/ultralytics#14186] [Q&A] [answered]
I'm using YOLOv8 as a detector for audio. I transform audios to mel spectograms and I have sounds I want to detect (no classification needed, so unique class is set to True during training for the model)

I want to create a logistic regression of the achieved output and I've read that it's better to do it with logits. As the output of YOLOv8 are confidence scores I would like to know if there is a recommended function to convert those confidence scores to logits in a post-processing step. SO, I already have the predictions of the BB with their scores, I just want to apply a function to those scores to transform them in logit score.

Thanks!

**Comment by pderrenger** [comment]
@GrunCrow hello,

Thank you for your question! It's great to hear that you're using YOLOv8 for audio detection through mel spectrograms. To transform confidence scores to logits, you can use the logit function, which is the inverse of the sigmoid function. The logit function can be defined as:

\[ \text{logit}(p) = \log\left(\frac{p}{1-p}\right) \]

Here's a simple Python function to convert confidence scores to logits:

```python
import numpy as np

def confidence_to_logit(confidence_scores):
    # Ensure the confidence scores are within the valid range (0, 1)
    confidence_scores = np.clip(confidence_scores, 1e-7, 1 - 1e-7)
    logits = np.log(confidence_scores / (1 - confidence_scores))
    return logits

# Example usage
confidence_scores = np.array([0.9, 0.8, 0.7])
logits = confidence_to_logit(confidence_scores)
print(logits)
```

This function will take your confidence scores and convert them to logits, which you can then use for your logistic regression.

Additionally, please ensure that you are using the latest version of the Ultralytics YOLO package to benefit from the latest features and bug fixes. If you encounter any issues, providing a minimum reproducible example can greatly help in diagnosing the problem. You can find more information on how to create one here: [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum_reproducible_example).

If you have any further questions or need additional assistance, feel free to ask. Happy coding! 😊

**Reply by GrunCrow** [reply]
Hi, thank you very much for the help, it worked!

**Reply by pderrenger** [reply]
Hello @GrunCrow,

You're very welcome! I'm glad to hear that the solution worked for you. 😊

If you have any more questions or run into any other issues, feel free to reach out. We're here to help!

Happy coding and best of luck with your audio detection project!

**Comment by glenn-jocher** [comment, marked as answer]
@GrunCrow to convert confidence scores to logits, you can use the formula: `logit = log(p / (1 - p))`, where `p` is the confidence score. This transformation can be applied in your post-processing step. Let me know if you need further assistance!

**Reply by GrunCrow** [reply]
Yes, thanks, Paula Derrenger replied to me some months ago and I came back to check the answer but I didn't find it. Thanks for replying again!

**Reply by glenn-jocher** [reply]
@GrunCrow you're welcome! If you have any more questions or need further assistance, feel free to ask.
